src/mainview.cpp

```diff
@@ -1,81 +1,60 @@
 #include "mainview.h"
 
-using namespace ui;
-
-// General list things
-
-struct MediaItem_t
-{
-	QString name;
-	QString icon;
-	QString urlType;
-	QString url;
-	QStringList args;
-};
-
-struct MediaSection_t
-{
-	QString header;
-	QVector<MediaItem_t> items;
-};
-
-static QVector<MediaSection_t> Sections(
-{
-	{
-		"Applications",
-		{
-			{ "Hammer World Editor", ":/resource/hammer.png", "process", "hammer.exe", { "" } },
-			{ "Model Viewer", ":/resource/modelviewer.png", "process", "hlmv.exe", { "-game p2ce" } },
-			{ "Face Poser", ":/resource/faceposer.png", "process", "hlfaceposer.exe", { "-game p2ce" } },
-			{ "P2:CE (Tools Mode)", ":/resource/logo_tools.png", "process", "chaos.exe", { "-game p2ce", "-tools" } },
-			{ "P2:CE", ":/resource/logo.png", "process", "chaos.exe", { "-game p2ce" } }
-		},
-	},
-	{
-		"Documentation",
-		{
-			{ "Valve Developer Community", ":/resource/vdc.png", "url", "https://developer.valvesoftware.com", { "" } },
-			{ "Chaos Wiki", ":/resource/chaos.png", "url", "https://chaosinitiative.github.io/Wiki/", { "" } },
-			{ "Momentum Wiki", ":/resource/momentum.png", "url", "https://docs.momentum-mod.org/", { "" } }
-		}
-	}
-});
+#include <QFile>
+#include <QJsonArray>
+#include <QJsonDocument>
+#include <QJsonObject>
 
+using namespace ui;
 
-CMainView::CMainView( QWidget *pParent ) : QDialog( pParent )
+CMainView::CMainView( QWidget *pParent ) :
+	QDialog( pParent )
 {
 	auto pLayout = new QVBoxLayout( this );
 	pLayout->setObjectName( "SDKLayout" );
 
-	for ( int i = 0; i < Sections.count(); i++ )
+	QFile config( "./config.json" );
```

Comment:
Couldn't we ultimately just load the default config in here rather than in `main.cpp`? Also this config path isn't really ideal. This would only work if `config.json` is next to the launcher.

Reply:
How would you have it? there is no real way to "search" for the config. it has to be stored somewhere.

Reply:
Would you like for me to move it to `mainview.cpp` and then have it use QDir::currentDir()? or how would you resolve the file's location?

Reply:
Just don't worry about it for now. I'll work on that when it comes to creating a project system for source mods.

Reply:
All right.